public/app/features/explore/state/explorePane.ts

```diff
@@ -258,8 +258,13 @@ export const paneReducer = (state: ExploreItemState = makeExplorePaneState(), ac
   }
 
   if (highlightLogsExpressionAction.match(action)) {
-    const { expressions } = action.payload;
-    return { ...state, logsHighlighterExpressions: expressions };
+    const { expressions: newExpressions } = action.payload;
+    const { logsHighlighterExpressions: currentExpressions } = state;
+
+    return {
+      ...state,
+      logsHighlighterExpressions: isEqual(newExpressions, currentExpressions) ? currentExpressions : newExpressions,
```

Comment:
Would just add a comment here that this is to prevent rerenders and that it comes from datasource so we cannot control if it returns new array or not.

Reply:
Added in https://github.com/grafana/grafana/pull/29752/commits/94eea0f47c81608a34af4874d990223ffde13347